### ISSUE TITLE
BoxTextured and BoxTexturedNonPowerOfTwo have different texture brightness

### ISSUE THREAD

**Issue by hybridherbst** [KhronosGroup/glTF-Sample-Models#345]
The screenshots of both models suggest that they are supposed to be identical, and also the textures seem to match:
https://github.com/KhronosGroup/glTF-Sample-Models/blob/master/2.0/BoxTexturedNonPowerOfTwo/glTF/CesiumLogoFlat.png
and
https://github.com/KhronosGroup/glTF-Sample-Models/blob/master/2.0/BoxTextured/glTF/CesiumLogoFlat.png.

Viewing the two files in any viewer shows that they have different brightness though.
When I view both files, the textures displayed in the texture view [gltf.report](https://gltf.report/) seem to match.
However, when viewing the individual textures in e.g. [Squoosh](https://squoosh.app), they differ.

Additionally, all viewers display these models with separate colors:
![image](https://user-images.githubusercontent.com/2693840/162057410-f7c0ff5a-d6a6-4a6e-bed3-1bbbe229470e.png)
_BoxTextured_
![image](https://user-images.githubusercontent.com/2693840/162057462-29d369e0-096f-4692-95bd-55821dfa9734.png)
_BoxTexturedNonPowerOfTwo_

My assumption is that at some point during image resizing, some software messed with the colors... now, from https://github.com/KhronosGroup/glTF-Sample-Models/pull/137 it looks like the non-power-of-two version actually came first.

On further checking, it turns out that while the non-power-of-two PNG doesn't contain a color profile, the power-of-two version contains one. I'm not sure but I don't think any viewer respects color profiles in PNG? That would explain why the textures look identical in gltf.report's texture viewer but not the viewport - the texture viewer uses HTML img which respects the profiles... 
EXIF data of the non-power-of-two version: https://www.metadata2go.com/result/2b7782e2-5962-412a-bd77-074f5f0e37ce
EXIF data of the power-of-two version: https://www.metadata2go.com/result/bca8acaa-2e17-44b3-9274-ab728209425f

Tested to look different in model-viewer, babylon, glTFast, UnityGLTF, Gestaltor, glTF Sample Viewer, Squoosh (textures only)

cc @donmccurdy for a number of reasons (color profiles, gltf.report)

**Comment by javagl** [comment]
Just as a pointer, https://www.khronos.org/registry/glTF/specs/2.0/glTF-2.0.html#file-extensions-and-media-types says

> PNG images SHOULD NOT contain animations, non-square pixel ratios, or embedded ICC profiles. Such features, if present, MUST be ignored by client implementations.

So **iff** (if and only if)  the profile is the only difference, then this MUST be ignored by the client, and thus, any visual difference could be considered as a bug in the viewer.

**Comment by hybridherbst** [comment]
Thanks for the pointer! I think it's the other way around though: the textures _do_ have a visual difference when only the raw bytes are considered and the ICC profile is not applied. 

The good thing: that means all viewers do the right thing (besides maybe the peculiarity that gltf.report shows the textures separately with ICC profile applied).

It also means that the bug here is that https://github.com/KhronosGroup/glTF-Sample-Models/blob/master/2.0/BoxTextured/glTF/CesiumLogoFlat.png has an ICC profile which is neither explained in the description nor I think intentional; there could be a separate file that explicitly contains an ICC profile in the PNG to test this.

**Comment by donmccurdy** [comment]
Correct, gltf.report uses HTML `<img/>` tags when displaying images in the texture browser tab. 

At one point I added a `ktxfix` command to gltf-transform to correct color profiles in KTX2 images. Early implementations had been writing some color profiles incorrectly at the time. I wonder if something similar would be useful for PNG/JPEG/WEBP, and whether it's easy to correct that metadata without external dependencies?

**Comment by hybridherbst** [comment]
With "correct the metadata", do you just mean "remove the color profile" (which would be incorrect I believe) or "recalculate the correct pixel values and remove the color profile"?
